packages/dashboard/app/withdraw/utils/transactionService.ts

```diff
@@ -0,0 +1,277 @@
+      };
+    } else {
+      return {
+        success: false,
+        hash: '',
+        error: error.message || 'Failed to send transaction'
+      };
+    }
+  }
+}
+
+/**
+ * Calculates the gas costs for ETH transactions on Base network
+ */
+export async function calculateEthGasCosts(): Promise<{
+  totalCost: ethers.BigNumber;
+  gasLimit: ethers.BigNumber;
+  gasPrice: ethers.BigNumber;
+}> {
+  const baseProvider = new ethers.providers.JsonRpcProvider(BASE_MAINNET_RPC);
+  const ethProvider = new ethers.providers.JsonRpcProvider(ETH_MAINNET_RPC);
+
+  // Get current gas prices for L1 and L2
+  const l1GasPrice = await ethProvider.getGasPrice();
+  const l2GasPrice = await baseProvider.getGasPrice();
+  
+  // Fixed gas limits
+  const l1GasLimit = ethers.BigNumber.from(1600);
+  const l2GasLimit = ethers.BigNumber.from(23100);
+  
+  // Calculate gas costs
+  const l1GasCost = l1GasLimit.mul(l1GasPrice);
+  const l2GasCost = l2GasLimit.mul(l2GasPrice);
+  const buffer = l1GasCost.add(l2GasCost).mul(50).div(100);
+  
+  // Total gas cost includes L1 fee, L2 gas, and buffer
+  const totalCost = l1GasCost.add(l2GasCost).add(buffer);
+  
+  const fixedBuffer = ethers.utils.parseEther('0.000001'); // 0.000001 ETH additional buffer
+  const finalCost = totalCost.add(fixedBuffer);
+  
+  return {
+    totalCost: finalCost,
+    gasLimit: l2GasLimit,
+    gasPrice: l2GasPrice
+  };
+}
+
+/**
+ * Sends an ETH transfer transaction
+ */
+export async function sendEthTransaction({
+  pkpWallet,
+  amount,
+  recipientAddress,
+  tokenBalance
+}: SendEthTransactionParams): Promise<TransactionResult> {
+  try {
+    const provider = new ethers.providers.JsonRpcProvider(BASE_MAINNET_RPC);
+
+    // Get gas info
+    const gasInfo = await calculateEthGasCosts();
+
+    // Check if amount is valid (the gas costs are already accounted for in handleMaxAmount)
+    if (amount.gt(tokenBalance)) {
+      return {
+        success: false,
+        hash: '',
+        error: `Insufficient balance for transaction.`
+      };
+    }
+
+    if (amount.lt(gasInfo.totalCost)) {
+      return {
+        success: false,
+        hash: '',
+        error: `Insufficient balance for gas fees.`
+      };
+    }
+
+    // Configure ETH transaction options
+    const txOptions = {
+      to: recipientAddress,
+      value: amount,
+      gasLimit: gasInfo.gasLimit,
+      gasPrice: gasInfo.gasPrice
+    };
+
+    console.log('Sending ETH transaction with options:', {
+      to: recipientAddress,
+      value: ethers.utils.formatEther(amount) + ' ETH',
+      gasLimit: txOptions.gasLimit.toString(),
+      gasPrice: ethers.utils.formatUnits(gasInfo.gasPrice, 'gwei') + ' gwei',
+      totalGasCost: ethers.utils.formatEther(gasInfo.totalCost) + ' ETH'
+    });
+
+    return await sendTransactionWithRetry({
+      pkpWallet,
+      txOptions,
+      provider,
+      initialGasPrice: gasInfo.gasPrice
+    });
```

Comment:
Following on the comment above, this function might double the gas cost which makes the previous check not valid anymore unless it considered that price increase